R/pretty_lollipop_plot.R

```diff
@@ -1,59 +1,219 @@
-#' @param out_name Optional, set the file name of the plot, if you export it to disk. Default name is my_lollipop_plot_{gene}.
-#'
-#' @return Nothing.
+#' @param plot_title Optional, the title of the plot. Default is gene.
+#' @param include_silent Logical parameter indicating whether to include silent mutations into coding mutations. Default is FALSE. 
+#' 
+#' @return A lollipop plot.
 #'
-#' @import g3viz dplyr
+#' @import dplyr ggplot2
 #' @export
 #'
 #' @examples
 #' library(GAMBLR.data)
 #' 
-#' #get metadata (Fl and DLBCL)
-#' metadata = get_gambl_metadata()
-#' this_meta = dplyr::filter(metadata, pairing_status == "matched")
-#' this_meta = dplyr::filter(this_meta, pathology %in% c("FL", "DLBCL"))
+#' #get meta data (BL_Thomas)
+#' metadata <- get_gambl_metadata() %>%
+#'     filter(cohort == "BL_Thomas")
 #' 
-#' maf = get_ssm_by_sample(these_samples_metadata = this_meta)
+#' maf_df <- get_ssm_by_samples(
+#'     these_samples_metadata = metadata
+#' )
 #'
 #' #construct pretty_lollipop_plot.
-#' pretty_lollipop_plot(maf_df = maf,
-#'                      gene = "MYC",
-#'                      plot_title = "Mutation data for MYC",
-#'                      plot_theme = "nature2")
+#' lolipop_result <- pretty_lollipop_plot(maf_df, "MYC")
 #'
-pretty_lollipop_plot = function(maf_df,
-                                gene,
-                                plot_title,
-                                plot_theme = "cbioportal",
-                                out_name = paste0("my_lollipop_plot_", gene)){
-  if(missing(gene)){
-    stop("Plese provide a gene...")
-  }
-
-  if(missing(plot_title)){
-    plot_title = gene
-  }
-
-  maf_df = maf_df %>%
-    dplyr::filter(Hugo_Symbol == gene)
-
-  #use the readMAF function (modified by Ryan) to parse/convert
-  maf_df = g3viz::readMAF(maf.df = maf_df)
-  chart.options = g3Lollipop.theme(theme.name = plot_theme, title.text = plot_title)
-  g3Lollipop(maf_df,
-             gene.symbol = gene,
-             plot.options = chart.options,
-             output.filename = out_name)
+pretty_lollipop_plot <- function(
+    maf_df, 
+    gene = NULL,
+    plot_title,
+    include_silent = FALSE
+) {
+    if(missing(gene)){
+        stop("Please provide a gene...")
+    }
+  
+    if(missing(plot_title)){
+        plot_title=gene
+    }
+
+    maf_df <- as.data.frame(maf_df)
+
+    # Specifying noncoding regions with coding_class as a bundled object with GAMBLR.helpers
+    if(include_silent){
+        variants <- coding_class
+    } else {
+        variants <- coding_class[!coding_class %in% c(
+            "Silent",
+            "Splice_Region"
+        )]
+    }
+
+    nc_maf_df <- maf_df %>%
+        filter(
+            Hugo_Symbol == gene
+        ) %>% 
+        filter(
+            Variant_Classification %in% variants 
+        ) 
+
+    # Filter for the specific gene 
+    gene_df <- nc_maf_df %>% 
+        filter(Hugo_Symbol == gene) %>%
```

Comment:
You already filtered for Hugo_Symbol == gene in line 62, so this filter is redundant and can be skipped